### DISCUSSION TITLE
[@types/node] Add types for node:test

### DISCUSSION THREAD

**Discussion by drazisil** [DefinitelyTyped/DefinitelyTyped#60607] [Issues with a @types package]
I would like to try out adding support for the [experimental test runner in Node.js v18](https://nodejs.org/api/test.html) 

There are a couple of questions I have:

* How, if at all, do I specify that is for v18?
* Since this is _only_ available under the `node:` import prefix, does it fit under https://github.com/DefinitelyTyped/DefinitelyTyped/tree/master/types/node at all, or does it require a completely new file path?

[a comment by an automated account is not shown]

**Comment by MysteryBlokHed** [comment]
I'm not an owner of the Node types, but I can try to answer your questions:

1. Since the current version of  `@types/node` is for version 17, I don't think there's a way to say that the module is only for version 18, other than maybe a mention in its JSDoc. You might have to wait until the types are updated for version 18, but an owner of the types will have to confirm

2. To specify that it's only available with the `node:` prefix, you can wrap the module's contents with `declare module 'node:test'`

`types/node/test.d.ts`
```typescript
/**
 * Test module
 * @since v18.0.0
 */
declare module 'node:test' {
    // module contents here
} 
```

**Comment by peterblazejewicz** [comment]
The types for Node are versioned within DT using versioning that is related how TS type acquisition works. So it's best to wait until current (top source) is at version 18:
https://github.com/DefinitelyTyped/DefinitelyTyped/pull/60047/files#diff-e40f0d452a99af4fb6e72ee34f613234bc536b74587a3db7f5b0ddcfdbb1ce29
One can start on this feature, just give it 'draft' status for now and watch #60047
